### PR TITLE
More documentation

### PR DESCRIPTION
**Pull request by googley42**

### CODE REVIEW

File: docs/concepts/architecture.md [hunk 1]
```diff
@@ -5,4 +5,12 @@ title: "Architecture"
 
 # Architecture
 
-TODO
+In the below diagram you can see the different layers of library and its dependencies.
```
Comment (googley42) [inline docs/concepts/architecture.md:8]: @khajavi  so the ZIO sbt `previewWebsite` task stopped working (website not running) and produces the following output in the logs (I am using the lasted version `0.4.0-alpha.28`)
```
> docusaurus start
sh: docusaurus: command not found
```
AFIAK management of docusaurus (eg install/setup) is done via the plugun. Any ideas?

Comment (khajavi) [reply docs/concepts/architecture.md:8]: clean the target directory and try `sbt installWebsite` before running `sbt previewWebsite`
@googley42 Please let me know if it didn't work!

Comment (googley42) [reply docs/concepts/architecture.md:8]: @khajavi  yes that worked - many thanks!

**Comment by googley42** [comment]
@khajavi so looks like the [Notify Docs Release GHA](https://github.com/zio/zio-dynamodb/actions/runs/12446491575/job/34748950809) is silently failing with a 401
```
Run PACKAGE_NAME=$(cat docs/package.json | grep '"name"' | awk -F'"' '{print $4}')
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   294  100   109  100   [18](https://github.com/zio/zio-dynamodb/actions/runs/12446491575/job/34748950809#step:3:19)5   2018   3425 --:--:-- --:--:-- --:--:--  5444
{
  "message": "Bad credentials",
  "documentation_url": "https://docs.github.com/rest",
  "status": "401"
}
```
Looks like this is the final [step](https://github.com/zio/zio-dynamodb/blob/series/2.x/.github/workflows/site.yml#L107-L134) that is preventing me from publishing to the main ZIO documentation website - I was wondering if you had any ideas?

**Pull request merged**